src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -89,10 +97,39 @@ class MoveModal extends React.Component {
     }
   }
 
-  moveEntry = (entryId, destinationId) => {
-    return this.context.client
-      .collection('io.cozy.files')
-      .updateFileMetadata(entryId, { dir_id: destinationId })
+  /**
+   * Try to move entry to destination.
+   *
+   * @param   {string} entryId       - The file's id
+   * @param   {string} destinationId - The target directory's id
+   * @param   {string} filename      - The file's name
+   * @returns {Promise}              - A promise that returns the move action response if resolved or an Error if rejected
+   *
+   */
+  moveEntry = async (entryId, destinationId, filename) => {
+    const { client } = this.context
+    try {
+      return await client
+        .collection('io.cozy.files')
+        .updateFileMetadata(entryId, { dir_id: destinationId })
+    } catch (e) {
+      if (flag('handle-move-conflicts') && e.status === 409) {
+        const targetPath = await CozyFile.getFullpath(destinationId, filename)
+        if (this.props.sharingState.sharedPaths.includes(targetPath)) {
+          throw e
+        } else {
+          const resp = await client
+            .collection('io.cozy.files')
+            .statByPath(targetPath)
+          await client.collection('io.cozy.files').destroy(resp.data)
+          return await client
+            .collection('io.cozy.files')
+            .updateFileMetadata(entryId, { dir_id: destinationId })
+        }
+      } else {
+        throw e
+      }
+    }
```

Comment:
Am I going to say it? Hmmm. Should we move this function to doctype?

Reply:
I thought about it this morning but I don't have a clear opinion on this. I'm not sure that we need a way to move a file except in `MoveModal`.
The main difficulty I see if we do this is that we need to be able to know which files have been deleted so that we can restore them in case of move cancel. I'm about to use the component's state to keep track of the files we delete while cancellation is possible.